dali/server/daldap.cpp

```diff
@@ -137,15 +137,24 @@ class CDaliLdapConnection: implements IDaliLdapConnection, public CInterface
         {
             username.append(filesdefaultuser);
             decrypt(password, filesdefaultpassword);
+            WARNLOG("Missing credentials, injecting deprecated filesdefaultuser");
         }
 
         Owned<ISecUser> user = ldapsecurity->createUser(username);
-        user->credentials().setPassword(password);
-        if (!ldapsecurity->authenticateUser(*user, NULL))
+
+        if (!isEmptyString(password))
         {
-            ERRLOG("LDAP: getPermissions(%s) scope=%s user=%s fails authentication",key?key:"NULL",obj?obj:"NULL",username.str());
-            return SecAccess_None;//deny
+            user->credentials().setPassword(password);
+            if (!ldapsecurity->authenticateUser(*user, NULL))
+            {
+                ERRLOG("LDAP: getPermissions(%s) scope=%s user=%s fails LDAP authentication",key?key:"NULL",obj?obj:"NULL",username.str());
+                return SecAccess_None;//deny
+            }
+            else
+                user->setAuthenticateStatus(AS_AUTHENTICATED);
```

Comment:
minor: Is this line needed?  I doubt it will cause problems, but probably unnecessary.

Reply:
I added that line because the following calls to authorizeFileScope/authorizeWorkunitScope call  CLdapSecManager::authenticate, which checks for that being set before attempting to call LDAP